libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerDialog2.cs

```diff
@@ -155,7 +173,11 @@ protected override Task<QnAMakerOptions> GetQnAMakerOptionsAsync(DialogContext d
             {
                 ScoreThreshold = this.Threshold,
                 StrictFilters = this.StrictFilters,
-                Top = this.Top
+                Top = this.Top,
+                Context = new QnARequestContext(),
+                QnAId = 0,
+                RankerType = RankerTypes.DefaultRankerType,
```

Comment:
>RankerTypes.DefaultRankerTyp [](start = 29, length = 28)

Shuoldn't this be this.RankerType?

Reply:
DefaultRankerType is const, so no need for this.

Reply:
You aren't using the declarative definition.  it should be
    THIS.RankerType


---
In reply to: [354976475](https://github.com/microsoft/botbuilder-dotnet/pull/3093#discussion_r354976475) [](ancestors = 354976475)